oracle-plugin/src/test/java/io/cdap/plugin/oracle/OracleSourceDBRecordUnitTest.java

```diff
@@ -95,6 +95,55 @@ public void validatePrecisionLessDecimalParsing() throws Exception {
     Assert.assertEquals(record.get("ID4"), "123.4568");
   }
 
+  /**
+   * Validate the precision less Numbers handling against following use cases.
+   * 1. Ensure that for Number(0,-127) non nullable type a String is returned by default.
+   * 2. Ensure that for Number(0,-127) non nullable type a String is returned,
+   *    if schema defined this as Number(38,4).
+   * 3. Ensure that for Number(0,-127) nullable type a String type is returned by default.
+   * 4. Ensure that for Number(0,-127) nullable type a String is returned,
+   *    if schema defined this as Number(38,4).
+   * @throws Exception
+   */
+  @Test
+  public void validatePrecisionLessNumberParsingForOutputSchemaAsDecimal() throws Exception {
+    Schema.Field field1 = Schema.Field.of("ID1", Schema.decimalOf(DEFAULT_PRECISION));
+    Schema.Field field2 = Schema.Field.of("ID2", Schema.decimalOf(DEFAULT_PRECISION, 4));
+    Schema.Field field3 = Schema.Field.of("ID3", Schema.nullableOf(Schema.decimalOf(DEFAULT_PRECISION)));
+    Schema.Field field4 = Schema.Field.of("ID4", Schema.decimalOf(DEFAULT_PRECISION, 4));
+
+    Schema schema = Schema.recordOf(
+      "dbRecord",
+      field1,
+      field2,
+      field3,
+      field4
+    );
+
+    when(resultSet.getMetaData()).thenReturn(resultSetMetaData);
+    when(resultSet.getBigDecimal(eq(1), eq(0))).thenReturn(new BigDecimal("123"));
```

Comment:
do we also have any tests where we actually read from an oracle DB? That would give more confidence than mocking the return value. For example, I see an OracleSourceTestRun, is that being executed?

Reply:
OracleSourceTestRun does not execute.